src/Integration/Laravel/Queue/src/AsyncAwsSqsQueue.php

```diff
@@ -0,0 +1,178 @@
+
+use AsyncAws\Illuminate\Queue\Job\AsyncAwsSqsJob;
+use AsyncAws\Sqs\Enum\QueueAttributeName;
+use AsyncAws\Sqs\SqsClient;
+use Illuminate\Contracts\Queue\Queue as QueueContract;
+use Illuminate\Queue\Queue;
+use Illuminate\Support\Str;
+
+class AsyncAwsSqsQueue extends Queue implements QueueContract
+{
+    /**
+     * The Amazon SQS instance.
+     *
+     * @var \AsyncAws\Sqs\SqsClient
+     */
+    protected $sqs;
+
+    /**
+     * The name of the default queue.
+     *
+     * @var string
+     */
+    protected $default;
+
+    /**
+     * The queue URL prefix.
+     *
+     * @var string
+     */
+    protected $prefix;
+
+    /**
+     * The queue name suffix.
+     *
+     * @var string
+     */
+    private $suffix;
+
+    /**
+     * Create a new Amazon SQS queue instance.
+     *
+     * @param string $default
+     * @param string $prefix
+     * @param string $suffix
+     *
+     * @return void
+     */
+    public function __construct(SqsClient $sqs, $default, $prefix = '', $suffix = '')
+    {
+        $this->sqs = $sqs;
+        $this->prefix = $prefix;
+        $this->default = $default;
+        $this->suffix = $suffix;
+    }
+
+    /**
+     * Get the size of the queue.
+     *
+     * @param string|null $queue
+     *
+     * @return int
+     */
+    public function size($queue = null)
+    {
+        $response = $this->sqs->getQueueAttributes([
+            'QueueUrl' => $this->getQueue($queue),
+            'AttributeNames' => [QueueAttributeName::APPROXIMATE_NUMBER_OF_MESSAGES],
+        ]);
+
+        $attributes = $response->getAttributes();
+
+        return (int) $attributes['ApproximateNumberOfMessages'];
+    }
+
+    /**
+     * Push a new job onto the queue.
+     *
+     * @param \Closure|string|object $job
+     * @param mixed                  $data
+     * @param string|null            $queue
+     *
+     * @return mixed
+     */
+    public function push($job, $data = '', $queue = null)
+    {
+        return $this->pushRaw($this->createPayload($job, $queue ?: $this->default, $data), $queue);
+    }
+
+    /**
+     * Push a raw payload onto the queue.
+     *
+     * @param string      $payload
+     * @param string|null $queue
+     *
+     * @return mixed
+     */
+    public function pushRaw($payload, $queue = null, array $options = [])
+    {
+        return $this->sqs->sendMessage([
+            'QueueUrl' => $this->getQueue($queue), 'MessageBody' => $payload,
+        ])->getMessageId();
+    }
+
+    /**
+     * Push a new job onto the queue after a delay.
+     *
+     * @param \DateTimeInterface|\DateInterval|int $delay
+     * @param \Closure|string|object               $job
+     * @param mixed                                $data
+     * @param string|null                          $queue
+     *
+     * @return mixed
+     */
+    public function later($delay, $job, $data = '', $queue = null)
+    {
+        return $this->sqs->sendMessage([
+            'QueueUrl' => $this->getQueue($queue),
```

Comment:
Are we sure this will return an URL? the part with `$prefix` is weird.

Reply:
I am fairly confident.